URLTester/Output/ProgressBar.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+
+
+
+
+namespace _URLTester.Output
+{
+    public interface IProgressBar
+    {
+        void UpdateProgressBar(int currentIndex, string currentItem);
+    }
+
+    public class ProgessBar : IProgressBar
+    {
+        const int totalBlocks = 30;
+        const int blockIncrement = 5;
+        const ConsoleColor blueConsole = ConsoleColor.Blue;
+        const ConsoleColor blackConsole = ConsoleColor.Black;
+
+        private readonly int _totalCount;
+
+        private int _currerntIndex;
+        private string _currentItem;
+
+        public ProgessBar(int totalCount)
+        {
+            _totalCount = totalCount;
+        }
+
+        public void UpdateProgressBar(int currentIndex, string currentItem)
+        {
+            _currerntIndex = currentIndex;
+            _currentItem = currentItem;
+
+
+            CreateBar();
+            UpdateBar();
+            DrawTotals();
+            DrawOutput();
+        }
+
+        private void DrawTotals()
+        {
+            Console.CursorLeft = totalBlocks + blockIncrement;
+            Console.BackgroundColor = blackConsole;
+        }
+
+        private void DrawOutput()
+        {
+            string output = _currerntIndex.ToString() + " of " + _totalCount.ToString();
+
+            var percentageComplete = CalculatePercenageComplete();
+            Console.Write(output.PadRight(15) + $"%{percentageComplete} \n {_currentItem}");
+        }
+
+        private int CalculatePercenageComplete()
+        {
+            return (int)(0.5f + ((100f * _currerntIndex) / _totalCount));
+        }
+
+        private void UpdateBar()
+        {
+            double pctComplete = Convert.ToDouble(_currerntIndex) / _totalCount;
+
+            int numChunksComplete = Convert.ToInt16(totalBlocks * pctComplete);
+
+            Console.BackgroundColor = blueConsole;
+            Console.Write("".PadRight(numChunksComplete));
+
+            Console.BackgroundColor = ConsoleColor.Gray;
+            Console.Write("".PadRight(totalBlocks - numChunksComplete));
+
+        }
+
+        private void CreateBar()
```

Comment:
This "Draw" method I created is an enhanced version of your CreateBar method that takes away the redundancy of rebuilding the progress bar on every single update.  Notice I changed the hardcoded Top and Left to a custom created struct called ConsolePoint that allows the programmer to pass in the X,Y coords: 

```suggestion
         public void Draw(ConsolePoint point)
        {
            Console.CursorTop = point.Y;
            Console.CursorLeft = point.X;
            Console.Write("[");
            Console.CursorLeft = Width + 1;
            Console.Write("]");
            Console.CursorLeft = point.X + 1;

            Console.BackgroundColor = ConsoleColor.Gray;
            Console.Write("".PadRight(Width));

            Console.CursorLeft = point.X + 1;
        }
```